lib/HALTarget/src/USBHost.cpp

```diff
@@ -233,6 +262,32 @@ size_t USBHost::readBytes(uint8_t* buffer, size_t length)
     return count;
 }
 
+void USBHost::requestBootloaderMode()
+{
+    /* Reset flags, ACM and data queue. */
+    reset();
+
+    /* Request bootloader mode. */
+    m_isBootloaderModeActive = true;
+    m_asyncOper.requestBootloaderMode();
+}
+
+void USBHost::reset()
+{
+    m_isBootloaderModeActive = false;
+    m_acm.Release();
+
+    if (nullptr != m_rxQueue)
+    {
+        (void)xQueueReset(m_rxQueue);
+    }
+}
+
+bool USBHost::isBootloaderModeActive() const
+{
+    return m_isBootloaderModeActive;

```

Comment:
How do you know that the device stays in the bootloader? It just returns that bootloader mode was requested.

Reply:
Every time the device leaves the bootloader mode, it triggers a timeout. When this happens, the flag and the connection are reset.

Reply:
This is definitely the "crude" way of doing it. It should be possible to create a state machine and switch the states based on the VID and PID of the USB device. These can only be obtained during the `process()` call, that is why the `m_isBootloaderModeActive` flag is required, but it would give us a better control over which device is connected and when

Reply:
Ok, I understood.